### PR TITLE
🗜 Working server using typescript

### LINKED ISSUE #8: Type checker
Thoughts on using flow?

### LINKED ISSUE #9: Document tech stack/choices
We don't have a package.json yet. We should document our decisions anyway.

### PR DESCRIPTION
**Pull request by astroash**
Changes:
- setup basic hapi server
- setup typescript
- update documentation to include tech choices and typescript install

Fixes #9 
Fixes #8

### CODE REVIEW

File: docs/stack.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+1. [Backend](#🛰-Backend)
+2. [Type Checker](#🗜-Type-Checker)
+3. [Database](#🗃-Database)
+4. [References](#🎓-References)
+
+## 🛰 Backend
+
+The server-side application is written in [Node.js](https://nodejs.org/en/) using the [Hapi js 17](https://hapijs.com/) framework. 
+
+🤷🏽‍ Hapi is ideal for an api serving multipe applications as it's designed for authentication stategies, and it's inbuild support for [joi](https://github.com/hapijs/joi) validation.
```
Comment (eliascodes) [inline docs/stack.md]: typos

File: docs/stack.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+4. [References](#🎓-References)
+
+## 🛰 Backend
+
+The server-side application is written in [Node.js](https://nodejs.org/en/) using the [Hapi js 17](https://hapijs.com/) framework. 
+
+🤷🏽‍ Hapi is ideal for an api serving multipe applications as it's designed for authentication stategies, and it's inbuild support for [joi](https://github.com/hapijs/joi) validation. 
+
+## 🗜 Type Checker
+[TypeScript](https://www.typescriptlang.org/) is used for static typing to offer .
```
Comment (eliascodes) [inline docs/stack.md]: incomplete sentence

File: docs/stack.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+## 🛰 Backend
+
+The server-side application is written in [Node.js](https://nodejs.org/en/) using the [Hapi js 17](https://hapijs.com/) framework. 
+
+🤷🏽‍ Hapi is ideal for an api serving multipe applications as it's designed for authentication stategies, and it's inbuild support for [joi](https://github.com/hapijs/joi) validation. 
+
+## 🗜 Type Checker
+[TypeScript](https://www.typescriptlang.org/) is used for static typing to offer . 
+
+🤷🏽‍ It was chosen over other options to due the open source type definitions for it's api.
```
Comment (eliascodes) [inline docs/stack.md]: it's -> its

**Comment by eliascodes** [comment]
Also rename the `server` directory to `src` please.

**Comment by eliascodes** [comment]
Also where is the command to build the server?

**Comment by astroash** [comment]
@eliascodes you run `tsc -w` as typescript recommended to be installed globally. Please see our docs for how to install this via npm.

I've not added start instructions as of yet, but can add it to this PR if you'd like.

**Comment by eliascodes** [comment]
Anything required in the normal course of development should at least be an `npm` script, and ideally also documented.

**Pull request merged**